bftengine/src/bftengine/BFTEngine.cpp

```diff
@@ -112,7 +112,7 @@ void ReplicaInternal::restartForDebug(uint32_t delayMillis) {
 namespace bftEngine {
 
 IReplica::IReplicaPtr IReplica::createNewReplica(const ReplicaConfig &replicaConfig,
-                                                 IRequestsHandler *requestsHandler,
+                                                 shared_ptr<IRequestsHandler> requestsHandler,
```

Comment:
Doesn't it make sense here to use unique_ptr? I think that request handler will never have shared ownership?